remodel/models.py

```diff
@@ -76,12 +76,13 @@ def save(self):
             result = (r.table(self._table).get(id_).replace(r.row
                         .without(r.row.keys().difference(list(fields_dict.keys())))
                         .merge(fields_dict), return_changes='always').run())
-
-        except KeyError:
+            if result['errors'] > 0:
```

Comment:
My use case is that I'm creating objects, and manually specifying an id (for example, migrating an existing database).  The objects I'm creating don't exist yet, so the update fails.  However, since the update isn't actually raising a KeyError, the insert code wasn't running.  My checking if there is an error in the result object, I can raise an exception manually.

Admittedly, I should be checking what the error is and handling it appropriately - maybe resorting to an insert, maybe something else.